app/models/service_discovery/cluster_client.rb

```diff
@@ -137,13 +139,22 @@ def find_resource(type, name, namespace = nil)
         resource_data = public_send("get_#{type.to_s.downcase}", *args)
         raise_not_found(type, name, namespace) unless resource_data
       rescue KubeException => exception
-        raise_not_found(type, name, namespace) if exception.error_code == 404
+        handle_kube_exception(exception, type, name, namespace)
       end
 
       klass = "ServiceDiscovery::Cluster#{type.to_s.camelize}".constantize
       klass.new(resource_data, self)
     end
 
+    def handle_kube_exception(exception, *args)
+      case exception.error_code
+      when 404
+        raise_not_found(*args)
+      else
+        raise ClusterClientError, exception.to_s
```

Comment:
Hmmm I do not think the cause error will give the error_code in Bugsnag
It would be good to have that reported somehow

Reply:
OK good enough. Might break in the future though if they decide to change the `to_s`